client/api/src/backend.rs

```diff
@@ -536,3 +536,22 @@ pub fn changes_tries_state_at_block<'a, Block: BlockT>(
 		None => Ok(None),
 	}
 }
+
+/// Provide CHT roots. These are stored on a light client and generated dynamically on a full
+/// client.
+pub trait ProvideChtRoots<Block: BlockT> {
+	/// Get headers CHT root for given block. Returns None if the block is not a part of any CHT.
```

Comment:
Iiuc this isn't how it works - it only returns `Some(_)` where CHTs are actually built (i.e. at headers 4097, 6145, ...), but for all other headers it'll return None (even though they may be part of already-built CHT).

Reply:
I've been looking at previous revision where `is_build_required` has been used, so nvm :)

Reply:
Yeah, the previous comment was correct for the light client, but not for the full client.